### PR TITLE
Changes for Embarcadero C++ clang-based compilers, targeting Boost 1.74. Change __BORLANDC__ to BOOST_BORLANDC, which is defined in Boost conf…

### PR DESCRIPTION
**Pull request by eldiener**
…ig for the Embarcadero non-clang-based compilers.

### CODE REVIEW

**Comment by eldiener** [comment]
Can this please be merged ?

File: include/boost/detail/catch_exceptions.hpp [hunk 1]
```diff
@@ -26,7 +26,7 @@
 #include <boost/cstdlib.hpp>  // for exit codes
 #include <ostream>         // for ostream
 
-# if defined(__BORLANDC__) && (__BORLANDC__ <= 0x0551)
+# if defined(BOOST_BORLANDC) && (BOOST_BORLANDC <= 0x0551)
```
Comment (Lastique) [inline include/boost/detail/catch_exceptions.hpp:29]: `#include <boost/config.hpp>` is missing. Same below.

Comment (eldiener) [reply include/boost/detail/catch_exceptions.hpp]: You are correct and I have updated the file in the PR. However the header file is using BOOST_NO_EXCEPTIONS, which definitely does require config.hpp, so should not boost/config.hpp be included ?

I also do not know what you mean by "Same below".

Comment (Lastique) [reply include/boost/detail/catch_exceptions.hpp]: No, my point is that you should add `#include <boost/config.hpp>`.

> I also do not know what you mean by "Same below".

I mean the other file you changed in the PR. It has the same issue of missing `boost/config.hpp`.

Comment (eldiener) [reply include/boost/detail/catch_exceptions.hpp]: Done !

**Pull request merged**